367 - Add getReadableOptions helper for options

nozomione opened the pull request:
## Issue Number

Epic: #366

Closes #367

Dependency (blocked by): #365

## Purpose/Implementation Notes

I've added the `getReadableOptions` helper and updated the implementations for Grommet `Select` and `ReadioButtonGroup` components, as well as cleaned up `config/options`.

Changes include:
- Added new helpers `getReadableOptions` and `uninqueArray` (for DRY)
- Added new properties to `config/readable` and cleaned up `config/options` (including the removal of duplicate definitions of `pageSizes` and  the `requestDataForm`)
- Updated the components to use `getReadableOptions`

Other changes:
- Added a type check to `gtag.TrackLink` to prevent en error (in 89dd3f386f6c262f7153f15dc1bc1acd66c0069b)

## Types of changes

- Bugfix (non-breaking change which fixes an issue)
- Refactor (addresses code organization and design mentioned in corresponding issue)
- New feature (non-breaking change which adds functionality)

## Functional tests
N/A

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Screenshots

N/A

nozomione commented on src/config/options.js:
```diff
@@ -116,6 +99,8 @@ export const options = {
       }
     ]
   },
+  // TODO: This will be cleaned up once the following PR is merged
+  // https://github.com/AlexsLemonade/refinebio-web/pull/365
```
As noted here, `transformation` will be removed once #365 is merged (it has already been removed in that PR).

nozomione commented on src/analytics/gtag.js:
```diff
@@ -101,7 +101,7 @@ const trackExploredUsageClick = (link) => {
 // tracks internal and external link clicks
 const trackLink = (link) => {
-  if (link.startsWith('http')) {
+  if (typeof link === 'string' && link.startsWith('http')) {
```
I've added this type check to prevent an error encountered during the development, thank you!

nozomione replied:
I mistakenly referred to `SearchCardFooter` instead of `SearchCardHeader`. Apologies for the confusion! I've now updated `SearchCardHeader` to ensure the `href` properly passes the formatted string and remove the type check for `link` from `gtag.trackLink`. Thank you David 👍!

nozomione commented on src/config/options.js:
```diff
@@ -48,29 +44,17 @@ export const options = {
+    page: 1
   },
+  // setting for the page size
+  pageSizes: [10, 20, 50],
```
Although this definition of the page size dropdown was identical, it was defined seperately in `search` and `samplesTable`, so I removed the duplicates.

nozomione commented on src/config/options.js:
```diff
@@ -48,29 +44,17 @@ export const options = {
       ARRAY_EXPRESS: 'ArrayExpress'
     }
   },
-  // setting for the request data form
-  requestDataForm: {
-    radioPediatricCancer: [
-      { label: 'Yes', value: 'Yes' },
-      { label: 'No', value: 'No' }
-    ],
-    radioPrimaryApproach: [
-      { label: 'Bench Research', value: 'Bench Research' },
-      { label: 'Computational Research', value: 'Computational Research' },
-      { label: 'Clinical Research', value: 'Clinical Research' },
-      { label: 'AI/ML Research', value: 'AI/ML Research' }
-    ]
-  },
```
I've cleaned up these definitions for the request data forms as both the labels and values are identical. They are no defined within the components and passed directly to Grommet components.

nozomione commented:
I've included the same change to `helpers/getRedable` aligning with the commit be8cc329255982685aa45b387f8c254429bdc895.

Additionally, I've applied 1:1 feedback to exclude `MoveToDatasetButton` from using `helpers/getReadableOptions`👍 

This PR is ready for another look, thank you David!

davidsmejia reviewed (approved):
One thing to address before merge and we can open a new issue to clean up how we pass data to child components.

nozomione commented:
I mistakenly referred to `SearchCardFooter` instead of `SearchCardHeader`. Apologies for the confusion! I’ve now updated `SearchCardHeader` to ensure the `href` prop properly passes the formatted string and removed the `link` parameter's type check in `gtag.trackLink` accordingly!

Now, I’m merging this PR into `dev`. Thank you, David! 👍
